magik-checks/src/main/java/nl/ramsolutions/sw/magik/checks/checks/NestingLevelCheck.java

```diff
@@ -0,0 +1,69 @@
+package nl.ramsolutions.sw.magik.checks.checks;
+
+import com.sonar.sslr.api.AstNode;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import nl.ramsolutions.sw.magik.checks.MagikCheck;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+
+/** Check nesting level of nodes. */
+@Rule(key = NestingLevelCheck.CHECK_KEY)
+public class NestingLevelCheck extends MagikCheck {
+
+  @SuppressWarnings("checkstyle:JavadocVariable")
+  public static final String CHECK_KEY = "NestingLevel";
+
+  private static final int DEFAULT_MAXIMUM_NESTING_LEVEL = 3;
```

Comment:
I would use the term depth instead of level. Also for the name of the check itself.